Sources/Vapor/HTTP/Headers/HTTPHeaders+ContentRange.swift

```diff
@@ -0,0 +1,263 @@
+                    return HTTPHeaders.Range.Value.from(requestStr: parameter)
+                }
+                return HTTPHeaders.Range.Value.from(requestStr: $0.1.value)
+            }
+            guard !rangeCandidates.isEmpty else {
+                return nil
+            }
+            self.ranges = rangeCandidates
+            let lowerCasedUnit = directives[0].value.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
+            self.unit = lowerCasedUnit == "bytes"
+                ? RangeUnit.bytes
+                : RangeUnit.custom(value: lowerCasedUnit)
+        }
+        
+        public func serialize() -> String {
+            return "\(unit.serialize())=\(ranges.map { $0.serialize() }.joined(separator: ", "))"
+        }
+    }
+    
+    /// Represents the HTTP `Content-Range` response header.
+    ///
+    /// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range
+    public struct ContentRange: Equatable {
+        public let unit: RangeUnit
+        public let range: HTTPHeaders.ContentRange.Value
+        
+        init?(directive: HTTPHeaders.Directive) {
+            let splitResult = directive.value.split(separator: " ")
+            guard splitResult.count == 2 else {
+                return nil
+            }
+            let (unitStr, rangeStr) = (splitResult[0], splitResult[1])
+            let lowerCasedUnit = unitStr.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
+            guard let contentRange = HTTPHeaders.ContentRange.Value.from(responseStr: rangeStr) else {
+                return nil
+            }
+            self.unit = lowerCasedUnit == "bytes"
+                ? RangeUnit.bytes
+                : RangeUnit.custom(value: lowerCasedUnit)
+            self.range = contentRange
+        }
+        
+        public init(unit: RangeUnit, range: HTTPHeaders.ContentRange.Value) {
+            self.unit = unit
+            self.range = range
+        }
+        
+        init?(directives: [Directive]) {
+            guard directives.count == 1 else {
+                return nil
+            }
+            self.init(directive: directives[0])
+        }
+        
+        public func serialize() -> String {
+            return "\(unit.serialize()) \(range.serialize())"
+        }
+        
+    }
+    
+    /// Convenience for accessing the Content-Range response header.
+    ///
+    /// See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range
+    public var contentRange: ContentRange? {
+        get {
+            let directives = parseDirectives(name: .contentRange).flatMap { $0 }
+            return HTTPHeaders.ContentRange(directives: directives)
+        }
+        set {
+            if contains(name: .contentRange) {
+                remove(name: .contentRange)
+            }
+            if newValue == nil {
+                return
+            }
+            add(name: .contentRange, value: newValue!.serialize())
+        }
+    }
```

Comment:
- Add a couple of `self`s to be in compliance with [the style guide](https://github.com/vapor/vapor/blob/master/.github/contributing.md)
- write getter in same style as for `range`
- use guard to avoid force unwrap
```suggestion
    public var contentRange: ContentRange? {
        get {
            return HTTPHeaders.ContentRange(directives: self.parseDirectives(name: .contentRange).flatMap { $0 })
        }
        set {
            if self.contains(name: .contentRange) {
                self.remove(name: .contentRange)
            }
            guard let newValue == newValue else {
                return
            }
            self.add(name: .contentRange, value: newValue.serialize())
        }
    }
```